app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -24,18 +24,20 @@ const TabCommonPropTypes = {
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
-const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
-}
+let tabTypeComponent = []
+
 
 const TopRibbon = ({tabs, routeProps}) =>
   <ul className={`tabs`}>
     {
       tabs.map((tab) =>
         <li title={tab.name} key={tab.type} className={`tabs-title`}>
+          {
+            tab.type === 'results' && Array.isArray(tab.props.ks) ? console.log("result1 **************", tabTypeComponent.push({'results' : TSnePlotViewRoute})) :
```

Comment:
@upendrakumbham Could you clean this code please? We are watching Uncle Bob's Clean Code videos to apply his teaching in our every day coding.
I would like you to find yourself why this code is not clean and I would like you to change it. It is not that hard.
If you struggle with it, then please ping me and I can help, but I would really like to see that you can do it yourself.
Thanks

Reply:
as per @lingyun1010 review comment, I have created JS functions to validate tabs and it's props.

Reply:
Please review and provide comments